File: 2106-find-greatest-common-divisor-of-array/2106-find-greatest-common-divisor-of-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    let maxNum=Math.max(...nums);
    let minNum=Math.min(...nums);
    function gcd(a, b) {
        while (b !== 0) {
            let temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    return gcd(maxNum, minNum);
};

File: 2106-find-greatest-common-divisor-of-array/2106-find-greatest-common-divisor-of-array.cs
using System;
using System.Linq;

public class Solution {
    public int FindGCD(int[] nums) {
        int maxNum = nums.Max(); 
        int minNum = nums.Min();

        return GCD(maxNum, minNum); 
    }

    private int GCD(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}
